openedx/adg/common/mailchimp_pipeline/helpers.py

```diff
@@ -23,3 +23,77 @@ def get_enrollment_course_names_and_short_ids_by_user(user):
 
     short_ids, display_names = zip(*enrollments)
     return ','.join(map(str, short_ids)), ','.join(display_names)
+
+
+def is_mailchimp_sync_required(created, sender, **kwargs):
+    """
+    Checks if sync is required with mailchimp. Sync will be done if `created` is `true`, `sender` is `UserProfile` or
+    `update_fields` have one of the fields which is required on mailchimp.
+
+    Args:
+        created (bool): True if the object of `sender` class is created
+        sender (obj): Class which sent the signal.
+        kwargs (dict): Additional Params.
+
+    Returns:
+        bool: True if sync is required else False.
+    """
+    MAILCHIMP_FIELDS = ['organization', 'status', 'business_line', 'company']
+    update_fields = kwargs.get('update_fields') or []
+    return created or sender == UserProfile or any(field in update_fields for field in MAILCHIMP_FIELDS)
+
+
+def get_user_merge_fields(instance):
```

Comment:
Now as we are sure the instance which it would be receiving is a user instance. So we can give it a specific name.

Reply:
Same goes for all these helpers.